src/Controller/DefaultController.php

```diff
@@ -114,8 +71,39 @@ private function getProjectDirectorList(array $leadershipPersons): array
      *
      * @return Response A Response instance.
      */
-    public function index()
+    public function index(FundingOrgFilter $fundingOrgFilter)
     {
+        $customTemplate = $this->getParameter('custom_template');
+
+        if ($customTemplate) {
+            $customTemplate = str_replace("base", "index", $customTemplate);
+            $filter = array();
+            if ($fundingOrgFilter->isActive()) {
+                $filter = array('fundingOrganization' => $fundingOrgFilter->getFilterIdArray());
+            }
+
+            $fundingCycles = $this->get('doctrine')->getRepository(FundingCycle::class)->findBy($filter, array('name' => 'ASC'));
+
+            $leadershipPersons = $this->get('doctrine')->getRepository(PersonResearchGroup::class)->getLeadershipPeople();
+
+            $projectDirectorList = $this->getProjectDirectorList($leadershipPersons);
+
+            $fundingCycleList = array();
+
+            foreach ($fundingCycles as $fundingCycle) {
+                $fundingCycleList[] = array(
+                    'id' => $fundingCycle->getId(),
+                    'name' => $fundingCycle->getName(),
+                    'researchGroups' => $this->getResearchGroupsArray($fundingCycle)
+                );
+            }
+
+            return $this->render("Default/$customTemplate", array(
```

Comment:
I figured out why there was an error, if this template doesn't exist, it breaks and doesn't even go to the default.
We should again rethink the switching mechanism, but switching by filter is not the same as switching by name/site/template.

Reply:
We can talk about this